File: staging_vespalib/src/vespa/vespalib/util/clock.cpp
// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.

#include "clock.h"
#include <cassert>
namespace vespalib {

Clock::Clock(const std::atomic<steady_time> & source) noexcept
    : _timeNS(source)
{
    static_assert(std::atomic<steady_time>::is_always_lock_free);
}

Clock::~Clock() = default;

}

File: clustercontroller-utils/src/main/java/com/yahoo/vespa/clustercontroller/utils/util/Clock.java
// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
package com.yahoo.vespa.clustercontroller.utils.util;

/**
 * Wrap access to clock so that we can override it in unit tests
 */
public class Clock {

    public long getTimeInMillis() { return System.currentTimeMillis(); }

    public int getTimeInSecs() { return (int)(getTimeInMillis() / 1000); }

}
